src/components/select-field/BaseSelectField.js

```diff
@@ -336,9 +336,20 @@ class BaseSelectField extends React.Component<Props, State> {
         }
     };
 
+    getItemFromFilteredOptions = (index: number) => {
+        const { options } = this.props;
+        const { searchText } = this.state;
+
+        const filteredOptions = options.filter(option =>
+            searchText ? option.displayText.toLowerCase().includes(searchText.toLowerCase()) : true,
+        );
+
+        return filteredOptions[index];
+    };
+
     selectSingleOption(index: number) {
-        const { options, selectedValues } = this.props;
-        const item = options[index];
+        const { selectedValues } = this.props;
+        const item = this.getItemFromFilteredOptions(index);
```

Comment:
This is the bulk of the fix. Before, we were setting item = options[index] but this logic will not work because index no longer refers to the correct index of the item that we just selected. Index now refers to the filtered index.

So for example, if we have options A, B, and C and then we perform a filter with the search string of 'C', when we click on C, the index that is being called in selectSingleOption is 0 because that is the only option being shown after the filter is applied. We then want to use the filtered list of options and index search in that instead.

When we want to get the correct item, we need to look through the filtered list of options and then apply the index to that.